base_tier_validation/models/tier_validation.py

```diff
@@ -211,7 +211,7 @@ def write(self, vals):
                 and not self._check_allow_write_under_validation(vals)
             ):
                 raise ValidationError(_("The operation is under validation."))
-        if vals.get(self._state_field) in self._state_from:
+        if vals.get(self._state_field) in [self._state_from, self._cancel_state]:
```

Comment:
First team is a list, so you have to merge both things:
```suggestion
        if vals.get(self._state_field) in (self._state_from + [self._cancel_state]):
```